articles/create-a-node-cli/index.md

```diff
@@ -0,0 +1,265 @@
+
+Using _process.arg_ might be tedious and not beginner friendly so we will use [yargs](https://www.npmjs.com/package/yargs) to help us process the options.
+
+Yargs offers a lot of options that make it very powerful and widely used. Some of them include `.command()`, `.help()`, `.example()`, `.usage()` and others. To read more on the different capabilities of the options see the yargs [docs](https://github.com/yargs/yargs).
+
+We will be making use of the command option to allow us to invoke a function when the set command is triggered. The syntax for this is:
+
+```javascript
+const argv = require("yargs");
+argv.command(
+  "name of command",
+  "description of the command",
+  functionA,
+  functionB
+).argv;
+```
+
+Note: _functionA_ will define the command arguments, then _functionB_ will implement the function.
+
+In our project we need two commands, let's set them up
+
+```javascript
+const argv = require("yargs");
+argv.command(
+  "joke",
+  "Fetching your joke",
+  (yargs) => {},
+  (argv) => {}
+).argv;
+
+argv.command(
+  "quote",
+  "Fetching a quote",
+  (yargs) => {},
+  (argv) => {}
+).argv;
+```
+
+Finally, let's pass in our functions to enable the jokes and quotes to be displayed.
+
+```javascript
+const argv = require("yargs");
+argv.command(
+  "joke",
+  "Fetching your joke",
+  (yargs) => {},
+  (argv) => {
+    getJoke();
+  }
+).argv;
+
+argv.command(
+  "quote",
+  "Fetching a quote",
+  (yargs) => {},
+  (argv) => {
+    getQuotes();
+  }
+).argv;
+```
+
+To get a joke or quote run:
+
+```terminal
+node index.js joke
+```
+
+```terminal
+node index.js quote
+```
+
+## Finishing Up
+
+Our full code now looks like this:
+
+```javascript
+#!/usr/bin/env node
+
+const axios = require("axios");
+const chalk = require("chalk");
+const argv = require("yargs");
+
+argv.command(
+  "joke",
+  "Fetching your joke",
+  (yargs) => {},
+  (argv) => {
+    getJoke();
+  }
+).argv;
+
+argv.command(
+  "quote",
+  "Fetching a quote",
+  (yargs) => {},
+  (argv) => {
+    getQuote();
+  }
+).argv;
+
+function getJoke() {
+  axios({
+    method: "get",
+    url: "https://official-joke-api.appspot.com/random_joke",
+  })
+    .then((res) => {
+      const setup = chalk.cyan(res.data.setup);
+      const punchline = chalk.green(res.data.punchline);
+      console.log(`${setup} - ${punchline}`);
+    })
+    .catch((err) => {
+      const log = chalk.red(err);
+      console.log(log);
+    });
+}
+
+function getQuote() {
+  axios({
+    method: "get",
+    url: "http://quotes.stormconsultancy.co.uk/random.json",
+  })
+    .then((res) => {
+      const randomQuote = chalk.cyan(res.data.quote);
+      const randomAuthor = chalk.green(res.data.author);
+      const log = `${randomQuote} - ${randomAuthor}`;
+      console.log(log);
+    })
+    .catch((err) => {
+      const errorLog = chalk.red(err);
+      console.log(errorLog);
+    });
+}
+argv.help();
```

Comment:
I ran this entire code snippet using `node index.js jokes` and nothing happened. Can you take a look at this?

Reply:
@louisefindlay23 the command as per the code for getting jokes is ```joke```, any other command will bring nothing in return.
Try, ```node index.js joke```

Reply:
Good catch, @katungi. Thanks.